### PR TITLE
Initial set of unit tests for static and local lamdbas

### PR DESCRIPTION
**Pull request by thk123**
Currently in a misnamed file `java_bytecode_parser_rclass_attribute.cpp`

### CODE REVIEW

File: src/java_bytecode/java_bytecode_parse_tree.h [hunk 1]
```diff
@@ -248,4 +248,21 @@ class java_bytecode_parse_treet
   }
 };
 
+/// Represents the argument of an instruction that uses a CONSTANT_Fieldref
+/// This is used for example as an argument to a getstatic and putstatic
+/// instruction
+class fieldref_exprt : public exprt
```
Comment (thk123) [inline src/java_bytecode/java_bytecode_parse_tree.h]: Note this is pulled into its own PR: diffblue/cbmc#1856

**Comment by thk123** [comment]
@mgudemann This is now ready for review and inclusion into the PR - I wonder if some of the tests are kind of redundant (e.g. we have the exhaustive static set, but then for member and local variable we just care about the ones that capture variables since they are the only ones that behave differently, what do you think?

File: unit/testing-utils/require_parse_tree.cpp [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+/// matching lambda entry found.
+/// \param parsed_class: the class to inspect
+/// \param descriptor: the descriptor the lambda method should have
+/// \param entry_index: the number to skip (i.e. if multiple entries with the
+/// same descriptor
+/// \return
+require_parse_tree::lambda_method_handlet
+require_parse_tree::require_lambda_entry_for_descriptor(
+  const java_bytecode_parse_treet::classt &parsed_class,
+  std::string descriptor,
```
Comment (mgudemann) [inline unit/testing-utils/require_parse_tree.cpp]: why not `const` ?

File: unit/testing-utils/require_parse_tree.cpp [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+  const size_t entry_index)
+{
+  REQUIRE(entry_index < parsed_class.lambda_method_handle_map.size());
+  typedef java_bytecode_parse_treet::classt::lambda_method_handle_mapt::
+    value_type lambda_method_entryt;
+
+  int matches_found = 0;
```
Comment (mgudemann) [inline unit/testing-utils/require_parse_tree.cpp]: `size_t` or `unsigned`

File: unit/testing-utils/require_parse_tree.cpp [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+
+  const auto matching_lambda_entry = std::find_if(
```
Comment (mgudemann) [inline unit/testing-utils/require_parse_tree.cpp]: we might also count to detect several matching entries

**Review by mgudemann — COMMENTED** [review]
looks good, just some minor comments
